hazelcast/src/main/java/com/hazelcast/cache/impl/CacheSplitBrainHandler.java

```diff
@@ -85,6 +86,11 @@ Runnable prepareMergeRunnable() {
                     }
                     // Clear all records either owned or backup
                     cacheRecordStore.clear();
+
+                    // send the cache invalidation event regardless if any actually cleared or not (no need to know how many
+                    // actually cleared)
+                    final CacheService cacheService = nodeEngine.getService(CacheService.SERVICE_NAME);
+                    cacheService.sendInvalidationEvent(cacheName, null, SOURCE_NOT_AVAILABLE);
```

Comment:
I put it so that the old behaviour is not changed. I am not very knowledgable on split brain handling, if it is not needed or we can handle it in a different way, yes I can change it.